src/common_fits.jl

```diff
@@ -20,18 +20,103 @@ function fit_line(xpts, ypts)
   return result.param, errors, fit_curve
 end
 
-function fit_ramsey(xpts, ypts)
-	model(t, p) = p[1]*exp(-t ./ p[2]).*cos(2π*p[3] .* t) + p[4]
+"""
+        fit_ramsey(xpts, ypts, yvars=[])
 
-	# Use KT estimation to get a guess for the fit
-	freqs,Ts,amps = KT_estimation(ypts, xpts[2]-xpts[1], 2)
-	idx = indmax(abs(amps))
+Fit data to a Ramsey decay of the form p[1]*exp(-t ./ p[2]).*cos(2π*p[3] .* t + p[4]) + p[5],
+with optional weights for each point.
+"""
+function fit_ramsey(xpts, ypts, yvars=[])
+    model(t, p) = p[1]*exp(-t ./ p[2]).*cos(2π*p[3] .* t + p[4]) + p[5]
 
-	p_guess = [abs(amps[idx]), Ts[idx], freqs[idx], mean(ypts)];
+    # Use KT estimation to get a guess for the fit
+    freqs,Ts,amps = KT_estimation(ypts-mean(ypts), xpts[2]-xpts[1], 1)
 
-	result = curve_fit(model, xpts, ypts, p_guess)
-	errors = estimate_errors(result)
-	return result.param[2:3], errors[2:3]
+    p_guess = [abs(amps[1]), Ts[1], freqs[1], angle(amps[1]), mean(ypts)];
+
+    if isempty(yvars)
+        result = curve_fit(model, xpts, ypts, p_guess)
+    else
+        result = curve_fit(model, xpts, ypts, 1./sqrt(yvars), p_guess)
+    end
+    errors = estimate_errors(result)
+    sq_error = sum(((model(xpts, result.param) - ypts).^2)./yvars)
+
+    dof = length(xpts)-5
+    Nσ = sq_error/sqrt(2dof) - dof/sqrt(2dof)
+
+    xfine = linspace(xpts[1],xpts[end],1001)
+    fit_curve = (xfine, model(xfine, result.param))
+
+    return result.param[2:3], sq_error, Nσ, errors[2:3], fit_curve
```

Comment:
I was thinking about something along these lines, but for the fit parameters -- it bothers me that I'd have to look at the source code to find out that the fit parameters mean.

What do you think of
```julia
immutable FitResult
    fit_params::Dict{String,Float64}
    sq_errror::Float64
    Nσ::Float64
    errors::Dict{String,Float64}
    fit_curve::Function
end
```
and then return something along the lines of
```julia
FitResult( Dict("A"=>p[1], "T"=>p[2], "f"=>p[3], "ϕ"=>p[4], "B"=>p[5]), ...
```
although the keys are not entirely "self-documenting".

Thoughts?